addons/web/static/tests/views/list_tests.js

```diff
@@ -277,6 +278,36 @@ QUnit.module('Views', {
         list.destroy();
     });
 
+    QUnit.test('editable list datetimepicker destroy widget', function (assert) {
+        assert.expect(2);
+        var done = assert.async();
+
+        var list = createView({
+            View: ListView,
+            model: 'foo',
+            data: this.data,
+            arch: '<tree editable="top">' +
+                    '<field name="date"/>' +
+                '</tree>',
+        });
+        list.$el.on({
+            'show.datetimepicker': function () {
+            assert.equal($('.bootstrap-datetimepicker-widget').length, 1,
+                'The datetimepicker is open');
+
+            list.$('input.o_datepicker_input').trigger($.Event('keydown', {which: $.ui.keyCode.ESCAPE}));
+
+            assert.equal($('.o_list_button_add').length, 1,
```

Comment:
i'd rather check that there is no row with className o_selected_row or something. Or the number of rows.

So i'd do a sanity check before clicking on Add, and the same check after Add and after pressing Escape.